resources/var/tmp/scm/init.script.d/040-install-plugins.groovy

```diff
@@ -30,6 +30,18 @@ def pluginsFromOldInstallation = []
 
 // methods
 
+def addMissingDefaultPluginsFromEtcd(plugins){
+    def defaultPlugins = getValueFromEtcd("scm_plugins_default").split(",");
```

Comment:
The key could be `null` and it is `null` in the most cases, because the etcd value is optional. This line is missing a null check, before split.